Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabase.cs

```diff
@@ -638,9 +660,29 @@ private Task ForceCacheFlushAsync(OperationContext context)
                 }).ThrowIfFailure();
         }
 
-        internal void ForceCacheFlush(OperationContext context)
+        internal bool ForceCacheFlush(OperationContext context, ContentLocationDatabaseCounters? counter = null, bool blocking = true)
         {
-            ForceCacheFlushAsync(context).GetAwaiter().GetResult();
+            // Ensure only one thread can start the flush
+            if (Interlocked.CompareExchange(ref _isFlushInProgress, 1, 0) == 1)

```

Comment:
After our discussion yesterday, I was going to suggest using SpinLock instead of lock for the inner comparison, but CompareExchange is even better. Well done improvement.